org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/AcquisitionDevice.java

```diff
@@ -277,24 +280,25 @@ public void run(IPosition parent) throws ScanningException, InterruptedException
 	        	if (!continueRunning) return;  // finally block performed 
 
 	        	// Run to the position
-        		manager.invoke(PointStart.class, pos);
-	        	positioner.setPosition(pos);   // moveTo in GDA8
+        		annotationManager.invoke(PointStart.class, pos);
+	        	positioner.setPosition(pos);          // moveTo in GDA8
+	        	exposureManager.setExposureTime(pos); // most of the time this does nothing.
 	        	
-	        	IPosition written = writers.await();          // Wait for the previous write out to return, if any
-	       		if (written!=null) manager.invoke(WriteComplete.class, written);
+	        	IPosition written = writers.await();  // Wait for the previous write out to return, if any
+	       		if (written!=null) annotationManager.invoke(WriteComplete.class, written);
 	        	
- 	        	runners.run(pos);              // GDA8: collectData() / GDA9: run() for Malcolm
-	        	writers.run(pos, false);       // Do not block on the readout, move to the next position immediately.
+ 	        	runners.run(pos);                     // GDA8: collectData() / GDA9: run() for Malcolm
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This line will not be executed conditionally; only the first line of this 3-line block will be. The rest will execute unconditionally. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2681)